### DISCUSSION TITLE
Demos on lona-web.org

### DISCUSSION THREAD

**Discussion by fscherf** [lona-web-org/lona#78] [Ideas]
Hi! We now have demos on https://lona-web.org/demos but at least a good demo for multi-user views is missing. Has someone a good idea for a view?

In my experience it's hard to explain the difference between multi-browser-tab (daemonizing) and multi-user (LonaView.iter_objects()). That's why i've chosen a chat view for test_project to illustrate multi-user views, because the the use-case is inherently multi-user. Maybe we add an polished version of that, or some kind of game maybe?

**Comment by maratori** [comment]
I've checked [daemonized view](https://lona-web.org/demos/daemonized-view/index.html). My first thought was that GIF shows two different browsers. But actually it shows two different tabs in the same browser. I've no idea how to improve it :(

Regarding multiplayer demo. Chat is a good example for me.
Another option is to implement [tic-tac-toe](https://en.wikipedia.org/wiki/Tic-tac-toe).

**Reply by fscherf** [reply]
@maratori I am not completely sure if i read Your comment correct. Do You think it needs improvement, but you don't know how?

tic-tac-toe is a great idea actually. Maybe we just do both

**Reply by maratori** [reply]
@fscherf I think that daemonized view demo (and documentation) is not clear enough. New users wouldn't understand the difference between ordinary view and daemonized one.

As far as I understand here is how they behave. Correct me if I'm wrong.

Ordinary view:
* new instance is created for each tab, even in the same browser (`handle_request` is called each time as well)
* instance is not killed after `handle_request` is returned
* instance is killed after tab is closed/reloaded (or even internet disconnected?)

Daemonized view:
* instance is created once for the browser
* instance is reused for all tabs in the browser (user will see exactly the same page in different tabs)
* instance is killed only after `handle_request` is returned
* instance is not killed after tab is closed/reloaded (user will see exactly the same page after tab reload)

All above works with default `LonaSessionMiddleware`.

**Reply by fscherf** [reply]
@maratori You are right. It seems to be confusing to anyone besides me, but i have no idea how to improve the API or the documentation

**Comment by fscherf** [comment]
FTR: lona-web.org has demos now
